packages/echo-base/src/errors/BaseError.test.ts

```diff
@@ -0,0 +1,68 @@
+import { BaseError, getAllProperties } from './BaseError';
+
+describe('BaseError', () => {
+    const message = 'this is a test';
+    it('should initialize with correct values', () => {
+        const actualError = new BaseError({ name: 'BaseError', message });
+        expect(actualError.message).toBe(message);
+        expect(actualError.name).toBe('BaseError');
+        expect(actualError.stack).toBeTruthy();
+    });
+
+    it('should preserve innerError as properties', () => {
+        const innerMessage = 'message';
+        const innerException = new CustomError(innerMessage);
+        const actualError = new BaseError({ name: 'BaseError', message, exception: innerException });
+        expect(actualError.message).toBe(message);
+        expect(actualError.name).toBe('BaseError');
+        expect(actualError.stack).toBeTruthy();
```

Comment:
I think this part is not needed, as this is already tested in the first test case.

Reply:
Corrected this one. Needs to be tested, to make sure we don't overwrite name/message in with properties.name/message.